inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -178,12 +179,23 @@ ie::CNNNetwork FrontEnd::convertNetwork(ie::CNNNetwork& network) {
         ngraph::opset5::Convolution::type_info,
         ngraph::opset5::GroupConvolution::type_info
     });
-
+    const std::vector<std::pair<ngraph::element::Type, ngraph::element::Type>> convert_precision_list {
+        {ngraph::element::i64, ngraph::element::i32},
+        {ngraph::element::u64, ngraph::element::i32},
+        {ngraph::element::u32, ngraph::element::i32},
+        {ngraph::element::u16, ngraph::element::i32},
+        {ngraph::element::boolean, ngraph::element::i32},
+    };
+    for (const auto &precision : convert_precision_list) {
+        manager.register_pass<ngraph::pass::ConvertPrecision>(precision.first, precision.second, myriad_type_to_fuse);
+    }
     manager.register_pass<vpu::DynamicToStaticShape>();
     manager.register_pass<vpu::EliminateShapeOfAfterDSR>();
     manager.register_pass<vpu::ConvertExtractImagePatchesToReorgYolo>();
     manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
     manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::i64, ngraph::element::i32, myriad_type_to_fuse);
```

Comment:
You need to put it after opset1 to legacy conversion as it produces a lot of i64 constants.

Reply:
Fixed.
But we need to have this pass before converting opset1 to legacy too due to some nodes have i64 precision after the conversions that were above (you can see an example of such a network in #-43654)

Reply:
This model is inferred fine on CPU.
In MKLDNN plugin ConvertPrecision passes are called after `ConvertOpSet2ToOpSet1`. Did you try to place them here?